userland/Makefile

```diff
@@ -1,20 +1,28 @@
 # userland master makefile. Included by application makefiles
 
-TOCK_USERLAND_BASE_DIR ?= $(abspath .)
-TOCK_BASE_DIR ?= $(abspath ../)
-BUILDDIR ?= $(abspath .)
+TOCK_USERLAND_BASE_DIR ?= ..
+TOCK_BASE_DIR ?= $(TOCK_USERLAND_BASE_DIR)/..
+BUILDDIR ?= build/$(TOCK_ARCH)
 TOCK_BOARD ?= storm
 TOCK_ARCH ?= cortex-m4
 LIBTOCK ?= $(TOCK_USERLAND_BASE_DIR)/libtock/build/$(TOCK_ARCH)/libtock.a
 
 TOOLCHAIN := arm-none-eabi
 
 # This could be replaced with an installed version of `elf2tbf`
-ELF2TBF ?= cargo run --manifest-path $(TOCK_USERLAND_BASE_DIR)/tools/elf2tbf/Cargo.toml --
+ELF2TBF ?= cargo run --manifest-path $(abspath $(TOCK_USERLAND_BASE_DIR))/tools/elf2tbf/Cargo.toml --
 ifdef PKG_NAME
 ELF2TBF_ARGS = -n $(PKG_NAME)
 endif
 
+# Files to compile. Only what is in the app directory.
+C_SRCS := $(wildcard *.c)
+CXX_SRCS := $(wildcard *.cc)
+OBJS += $(patsubst %.c,$(BUILDDIR)/%.o,$(C_SRCS))
+OBJS += $(patsubst %.cc,$(BUILDDIR)/%.o,$(CXX_SRCS))
+
+CPPFLAGS += -DSTACK_SIZE=2048
```

Comment:
Well if we're going to do this we may as well change the default stack size instead